chain/chain/src/chain.rs

```diff
@@ -2370,6 +2370,30 @@ impl Chain {
     pub fn is_chunk_orphan(&self, hash: &CryptoHash) -> bool {
         self.blocks_with_missing_chunks.contains(hash)
     }
+
+    /// Check if can sync with sync_hash
+    pub fn sync_hash_validity(&mut self, sync_hash: &CryptoHash) -> Result<bool, Error> {
+        let head = self.head()?;
+        match self.get_block_header(sync_hash) {
+            Ok(sync_header) => {
+                let sync_header = sync_header.clone();
```

Comment:
I don't think you need to clone here. You can do something like `let prev_hash = sync_header.prev_hash()` to get around the borrow checker.

Reply:
++